x/ethbridge/keeper/msg_server.go

```diff
@@ -146,31 +146,56 @@ func (srv msgServer) CreateEthBridgeClaim(goCtx context.Context, msg *types.MsgC
 		}
 	}
 
+	claim := msg.EthBridgeClaim
+
+	metadata := types.TokenMetadata{
+		Decimals:          claim.Decimals,
+		Name:              claim.TokenName,
+		Symbol:            claim.Symbol,
+		TokenAddress:      claim.TokenContractAddress,
+		NetworkDescriptor: claim.NetworkDescriptor,
+	}
+
+	if !srv.Keeper.ExistsTokenMetadata(ctx, claim.DenomHash) {
+		// If this is the first time we see this token its not
+		// double pegged correct?
+		metadataMapping := types.TokenMetadataMapping{
+			DoublePegged:  false,
+			SourceAddress: claim.TokenContractAddress,
+		}
+		networkMapping := make(map[string]*types.TokenMetadataMapping)
+		networkMapping[claim.NetworkDescriptor.String()] = &metadataMapping
```

Comment:
Why is this needed?

Reply:
This code is checking if a token that a claim is made against currently has its metadata stored in the metadata module. currently nothing else adds or checks this data, but its stored if not currently available. It was unclear to me how data would otherwise be stored into this module.

Reply:
I understand that we are putting the token metadata into the keeper if it doesn't already exist, there, my question is, why is the networkMapping needed and what is its usecase?

Reply:
Network Mapping is defined in the Coin Metadata Conventions document; while it wasn't exactly clear to me how it was supposed to be used, I was trying to implement it according to this description:

> Denom => {
> 	decimals: 9
> 	name: “wBTC”
> 	symbol: “WBTC”
> 	tokenAddress: “0xbF45BFc92ebD305d4C0baf8395c4299bdFCE9EA2”
> 	networkDescriptor: “2”
> 		networkDescriptor => {
> doublePegged: true
> }
> }
> 
> The metadata included in the keeper will be the smart contract address, the decimals, the network descriptor, token name and symbol. When relayers submit a transaction to peg an asset into sifchain, they will send the token address, network descriptor, decimal amount, token name, and token symbol. Then, if that token has not already been pegged into sifchain, the token address, name, symbol, network descriptor and decimal amount will be stored in the metadata module. There will also be a nested mapping that maps a network descriptor to the smart contract address on the source chain.

Reply:
We only need to modify that mapping when we double peg an asset from sifchain to an external chain. This is a change that you will need to make to the lock function to support double pegging.